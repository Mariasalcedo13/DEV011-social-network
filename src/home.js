function home(navigateTo) {
  const mainPage = document.createElement('div');
  mainPage.setAttribute('class', 'homepage1');

  const containerimg = document.createElement('div');
  containerimg.className = 'containerimg';
  const img = document.createElement('img');
  img.src = 'img/planta-arana.png';
  img.id = 'homeplant';
  img.alt = 'Imagen de plantas';
  const content = document.createElement('div');
  content.className = 'content';

  const homewelcome = document.createElement('h3');
  homewelcome.className = 'homewelcome';
  homewelcome.textContent = 'Bienvenidos a :';

  const hometitle = document.createElement('h1');
  hometitle.className = 'hometittle';
  hometitle.textContent = 'Mi PlantApp';

  const words = document.createElement('p');
  words.className = 'words';
  words.textContent = 'Una gran comunidad de amantes de las plantas y botánica.';
  const orderbuttons = document.createElement('div');
  orderbuttons.className = 'orderbuttons';

  const registerButton = document.createElement('button');
  registerButton.className = 'registrationButton';
  registerButton.id = 'registerButton';
  registerButton.textContent = 'Registrarse';

  registerButton.addEventListener('click', () => {
    navigateTo('/register');
  });

  const wordsHome = document.createElement('p');
  wordsHome.className = 'wordsHome';
  wordsHome.textContent = '¿Ya tienes una cuenta?';

  const loginButton = document.createElement('button');
  loginButton.className = 'loginButton';
  loginButton.id = 'loginButton';
  loginButton.textContent = 'Iniciar Sesión';
  loginButton.addEventListener('click', () => {
    navigateTo('/login');
  });

  mainPage.appendChild(containerimg);
  containerimg.appendChild(img);
  mainPage.appendChild(content);
  content.appendChild(homewelcome);
  content.appendChild(hometitle);
  content.appendChild(words);
  content.appendChild(orderbuttons);
  orderbuttons.appendChild(registerButton);
  orderbuttons.appendChild(wordsHome);
  orderbuttons.appendChild(loginButton);
  return mainPage;
}

export default home;
