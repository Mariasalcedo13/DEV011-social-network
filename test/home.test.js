/**
 * @jest-environment jsdom
 */
import home from '../src/home.js';

describe('home', () => {
  it('Should be a function', () => {
    expect(typeof home).toBe('function');
  });
  it('Have a button to Login', () => {
    const DOM = document.createElement('div');
    DOM.append(home());
    const haveAbutton = DOM.querySelector('#loginButton');
    expect(haveAbutton).not.toBe(undefined);
  });
  it('Have a button to register', () => {
    const DOM = document.createElement('div');
    DOM.append(home());
    const haveAbutton = DOM.querySelector('#registerButton');
    expect(haveAbutton).not.toBe(undefined);
  });
  it('After click button return call function navigateTo with /login', () => {
    const DOM = document.createElement('div');
    const navigateTo = jest.fn();
    DOM.append(home(navigateTo));
    const haveAbuttonLogin = DOM.querySelector('#loginButton');
    haveAbuttonLogin.click();
    expect(navigateTo).toHaveBeenLastCalledWith('/login');
  });
  it('After click button return call function navigateTo with /register', () => {
    const DOM = document.createElement('div');
    const navigateTo = jest.fn();
    DOM.append(home(navigateTo));
    const haveAbuttonRegister = DOM.querySelector('#registerButton');
    haveAbuttonRegister.click();
    expect(navigateTo).toHaveBeenLastCalledWith('/register');
  });
  it('Snapshot of home', () => {
    const DOM = document.createElement('div');
    DOM.append(home());
    expect(DOM).toMatchSnapshot();
  });
});
