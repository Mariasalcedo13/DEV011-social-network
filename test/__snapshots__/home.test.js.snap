// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home Snapshot of home 1`] = `
<div>
  <div
    class="homepage1"
  >
    <div
      class="containerimg"
    >
      <img
        alt="Imagen de plantas"
        id="homeplant"
        src="img/planta-arana.png"
      />
    </div>
    <div
      class="content"
    >
      <h3
        class="homewelcome"
      >
        Bienvenidos a :
      </h3>
      <h1
        class="hometittle"
      >
        Mi PlantApp
      </h1>
      <p
        class="words"
      >
        Una gran comunidad de amantes de las plantas y botánica.
      </p>
      <div
        class="orderbuttons"
      >
        <button
          class="registrationButton"
          id="registerButton"
        >
          Registrarse
        </button>
        <p
          class="wordsHome"
        >
          ¿Ya tienes una cuenta?
        </p>
        <button
          class="loginButton"
          id="loginButton"
        >
          Iniciar Sesión
        </button>
      </div>
    </div>
  </div>
</div>
`;
